### PR TITLE
Mappool improvements

### PR DESCRIPTION
**Pull request by 7mochi**

### CODE REVIEW

File: app/commands.py [hunk 1]
```diff
@@ -2277,17 +2278,19 @@ async def mp_pick(ctx: Context, match: Match) -> Optional[str]:
     match.map_id = bmap.id
     match.map_name = bmap.full_name
 
-    # TODO: some kind of abstraction allowing
-    # for something like !mp pick fm.
-    if match.freemods:
-        # if freemods are enabled, disable them.
+    # TODO: Improve this spaghetti code
+    if mods == MappoolMods.FREEMODS or mods == MappoolMods.TIEBREAKER:
+        match.freemods = True
```
Comment (cmyui) [inline app/commands.py]:
```suggestion
    if mods in (MappoolMods.FREEMODS, MappoolMods.TIEBREAKER):
        match.freemods = True
```

File: app/constants/mappool_mods.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    NOMOD = 0
+    EASY = 1 << 1
+    HIDDEN = 1 << 3
+    HARDROCK = 1 << 4
+    DOUBLETIME = 1 << 6
+    FLASHLIGHT = 1 << 10
+    FREEMODS = 1 << 31
+    TIEBREAKER = 1 << 32
+
+    @functools.cache
```
Comment (cmyui) [inline app/constants/mappool_mods.py]: functools.cache on methods causes memory leaks
https://www.youtube.com/watch?v=sVjtp6tGo0g
```suggestion
```

File: app/commands.py (hunk 1 above)
Comment (cmyui) [reply app/commands.py]: not really spaghetti code, i think it's fine

File: app/constants/mappool_mods.py (hunk 2 above)
Comment (7mochi) [reply app/constants/mappool_mods.py]: Thanks!

**Pull request merged**